src/pages/Wallet/AddressDetailsPage.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AnimatePresence } from 'framer-motion'
+import { ArrowLeft, Settings as SettingsIcon } from 'lucide-react'
+import { useState } from 'react'
+import { useHistory, useParams } from 'react-router-dom'
+import styled from 'styled-components'
+
+import Amount from '../../components/Amount'
+import Button from '../../components/Button'
+import ClipboardButton from '../../components/Buttons/ClipboardButton'
+import OpenInExplorerButton from '../../components/Buttons/OpenInExplorerButton'
+import QRCodeButton from '../../components/Buttons/QRCodeButton'
+import DataList, { DataListCell, DataListRow } from '../../components/DataList'
+import Label from '../../components/Label'
+import { MainContent, PageTitleRow } from '../../components/PageComponents/PageContainers'
+import { PageH1, PageH2 } from '../../components/PageComponents/PageHeadings'
+import Table, { TableProps } from '../../components/Table'
+import { useAddressesContext } from '../../contexts/addresses'
+import AddressOptionsModal from './AddressOptionsModal'
+
+const transactionsTableHeaders: TableProps['headers'] = [
+  { title: 'Direction' },
+  { title: 'Timestamp' },
+  { title: 'Address(es)' },
+  { title: 'Amount', align: 'end' }
+]
+
+const AddressDetailsPage = () => {
+  const [isAddressOptionsModalOpen, setIsAddressOptionsModalOpen] = useState(false)
+  const { addressesState } = useAddressesContext()
+  const { address } = useParams<{ address: string }>()
```

Comment:
Tiny detail related to my comment in the address context file, we could use the `AddressHash` type for full explicitness.